maven/impl-maven/src/test/java/org/jboss/shrinkwrap/resolver/impl/maven/bootstrap/SystemPropertyPrecedenceTestCase.java

```diff
@@ -79,24 +79,21 @@ public void overrideGlobalSettings() {
 
     }
 
-    @Test(expected = NoResolvedResultException.class)
-    public void overrideOfflineFlag() {
-
-        try {
-            System.setProperty(MavenSettingsBuilder.ALT_MAVEN_OFFLINE, "true");
+    @Test
+    void overrideOfflineFlag() {
+        System.setProperty(MavenSettingsBuilder.ALT_MAVEN_OFFLINE, "true");
```

Comment:
The problem I see in this change is that in case of a test failure the property is still set and might affect other tests either in this testsuite or somewhere else (as it's a system property). I'd propose to change this testsuite in this PR and create a proper cleanup method AfterEach